### PR TITLE
copy build labels from buildconfig

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@csrwng ptal.
@jwforres fyi.

**Comment by bparees** [comment]
[test]

**Comment by csrwng** [comment]
LGTM

**Comment by bparees** [comment]
[merge]

File: pkg/build/util/generate.go [hunk 1]
```diff
@@ -27,9 +27,14 @@ func GenerateBuildFromConfig(bc *buildapi.BuildConfig, r *buildapi.SourceRevisio
 			Revision: r,
 		},
 		ObjectMeta: kapi.ObjectMeta{
-			Labels: map[string]string{buildapi.BuildConfigLabel: bcCopy.Name},
+			Labels: bcCopy.Labels,
```
Comment (smarterclayton) [inline pkg/build/util/generate.go]: You need to be copying this map, not referencing it.

Comment (bparees) [inline pkg/build/util/generate.go]: why? the buildconfig is already a copy explicitly made for this purpose...

Comment (smarterclayton) [inline pkg/build/util/generate.go]: Is the build config thrown away?  What if we start caching build configs and its coming out of the cache?  We already hit this a few times in Kube, so I'm a bit paranoid about it:

Comment (bparees) [inline pkg/build/util/generate.go]: you can see the whole scope of bcCopy here.  The copy is created in this function and not returned out of it.  The only reason the copy is being made is so we can assign fields from it into the build we are creating w/o them sharing data.

Comment (smarterclayton) [inline pkg/build/util/generate.go]: K

**Comment by bparees** [comment]
[merge]

**Pull request merged**